src/validation.cpp

```diff
@@ -2023,15 +2026,12 @@ void Chainstate::CheckForkWarningConditions()
 {
     AssertLockHeld(cs_main);
 
-    // Before we get past initial download, we cannot reliably alert about forks
-    // (we assume we don't get stuck on a fork before finishing our initial sync)
-    // Also not applicable to the background chainstate
-    if (m_chainman.IsInitialBlockDownload() || this->GetRole() == ChainstateRole::BACKGROUND) {
+    if (this->GetRole() == ChainstateRole::BACKGROUND) {
         return;
     }
 
     if (m_chainman.m_best_invalid && m_chainman.m_best_invalid->nChainWork > m_chain.Tip()->nChainWork + (GetBlockProof(*m_chain.Tip()) * 6)) {
-        LogPrintf("%s: Warning: Found invalid chain at least ~6 blocks longer than our best chain.\nChain state database corruption likely.\n", __func__);
+        LogWarning("Found invalid chain at least ~6 blocks longer than our best chain. Chain state database corruption likely.");
```

Comment:
nit: This message is arguably off-by-one compared to the test above.

Only `m_chainman.m_best_invalid->nChainWork > m_chain.Tip()->nChainWork` would be:

```
Found invalid chain at least ~1 blocks longer than our best chain. Chain state database corruption likely.
```

```suggestion
        LogWarning("Found invalid chain at least ~7 blocks longer than our best chain. Chain state database corruption likely.");
```
Noticed this when writing the test.